gateway-provider-identity-assertion-common/src/main/java/org/apache/knox/gateway/identityasserter/common/filter/AbstractIdentityAssertionFilter.java

```diff
@@ -110,7 +111,9 @@ protected void continueChainAsPrincipal(HttpServletRequestWrapper request, Servl
         if (primaryPrincipal != null) {
           if (!primaryPrincipal.getName().equals(mappedPrincipalName)) {
             impersonationNeeded = true;
-            auditService.getContext().setProxyUsername( mappedPrincipalName );
+            AuditContext context = auditService.getContext();
+            context.setProxyUsername( mappedPrincipalName );
+            auditService.attachContext(context);
```

Comment:
In log4j1 the `MDC` held a reference to an object, so changing the `proxyUserName` on this object changed the `MDC` as well. But in log4j2 there is a `ThreadContext` instead of `MDC` which can hold only key value pairs. So changing the `proxyUserName` on the `AuditContext` won't change the the threadContext without reattaching (thus populating the key values) again.

Reply:
Ack